src/lib/idp/introspectToken.ts

```diff
@@ -1,30 +1,36 @@
 import { SignJWT } from "jose";
-import axios from "axios";
+import axios, { type AxiosError } from "axios";
 // biome-ignore lint/correctness/noNodejsModules: we need the node crypto module
-import crypto from "node:crypto";
+import { createPrivateKey } from "node:crypto";
 import { ZITADEL_APPLICATION_KEY, ZITADEL_DOMAIN } from "@src/config";
 
-export async function introspectToken(token: string) {
-  const alg = "RS256";
-  const privateKey = crypto.createPrivateKey({
-    key: JSON.parse(ZITADEL_APPLICATION_KEY).key,
-  });
-  const clientId = JSON.parse(ZITADEL_APPLICATION_KEY).clientId;
-  const kid = JSON.parse(ZITADEL_APPLICATION_KEY).keyId;
+type IntrospectionResult = {
+  username: string;
+  exp: number;
+};
 
+const algorithm = "RS256";
```

Comment:
With optimization in mind I moved some of the variables out of the `introspectToken` function as they can be initialized once for all future calls to `introspectToken`.